### PR TITLE
feat: add limen profile command, profiler module, and run startup summary

### PR DESCRIPTION
**Pull request by pdey**
**NOTE:** The author must always **first review the PR themselves**, before requesting review from others, that means carefully having reviewed the full diff in GitHub.

## What does this PR change?

- Add `limen profile` CLI command — profiles a YAML experiment's permutation space (total count, per-parameter cardinalities, complexity rating) and, when `test_data_source` is configured, runs a randomised strength-1 covering array to estimate per-permutation runtime
- Add `limen/yaml/profiler.py` with `ProfileResult`, `make_covering_array`, and `profile()` — profiler core usable independently of the CLI

Implements experiment profiler phase of #506 

## Checklist

- [x] I have reviewed full diff in “Files changed”
- [x] I left no unnecessary files in the changes
- [x] I ran `python tests/run.py` locally without errors (where applicable)
- [ ] I updated `/docs` (if behavior/API/config/user/etc changed)
- [x] I added and/or updated docstrings as per [Writing Docstrings](https://github.com/Vaquum/Limen/blob/main/docs/Developer/Writing-Docstrings.md) (for any changed public functions/classes)
- [x] I updated CHANGELOG.md (unless only docs or other non-code aspect was changed)
- [x] I updated pyproject.toml (unless only docs or other non-code aspect was changed)
- [x] I added and/or updated tests (if behavior changed or new code paths added)
- [x] I validated changes manually
- [x] I validated changes with LLM
- [x] I removed any extraneous examples/comments
- [x] I linked issue to auto-close on merge (e.g., “Fixes #123”) when applicable

### CODE REVIEW

File: limen/yaml/profiler.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    return [{key: columns[key][i] for key in params} for i in range(n)]
+
+
+def _classify_error(exc: Exception) -> str:
+
+    msg = str(exc)
+    lower = msg.lower()
+    if 'sample' in lower and ('0 ' in lower or 'empty' in lower):
+        return f'Test data too small — increase n_rows in test_data_source. ({msg})'
+    if 'class' in lower and ('one' in lower or '1' in lower or 'least' in lower):
```
Comment (zero-bang) [inline limen/yaml/profiler.py]: The `'1' in lower` check is too broad — it matches the single character `'1'` anywhere in the error message. Combined with the `'class'` check, this misclassifies common ML errors as class-diversity issues:

```python
_classify_error(ValueError('XGBClassifier: learning_rate must be > 0.1'))
# → 'Not enough class diversity in test data — try larger n_rows.'

_classify_error(RuntimeError('LogisticRegression classification failed after 1 iteration'))
# → 'Not enough class diversity in test data — try larger n_rows.'
```

The `'least'` and `'one'` branches already cover sklearn's actual class-imbalance messages ("at least 2", "only one class"). Consider dropping `'1' in lower` or replacing it with something tighter like `'only 1' in lower`.

Same family of issue on line 87: `'inf' in lower` matches "information" → `_classify_error(ValueError('For more information see sklearn docs'))` returns "Test data contains NaN or Inf values."

The original error is always preserved in parentheses so the impact is cosmetic, but the misleading advice ("try larger n_rows") for unrelated errors could send users on a wild goose chase.

**Review by zero-bang — APPROVED** [review]
Well-structured PR. The `_load_yaml.py` extraction is a clean DRY improvement, the profiler design is solid (covering array + timing separation), and the CLI integration is well done. Test coverage for the new code is thorough — 34 new tests covering edge cases, partial failures, and mocks.

One non-blocking issue: `_classify_error` has overly broad substring matching that produces false positives on realistic ML errors. See inline comment.

The coverage CI drop (84% → 42%) is noise from unrelated files (indicators, backtest, trading) — the new code itself is well-tested.
